## Exploratory Data Analysis (EDA)

### 1. Load Data and Initial Inspection

This first step involves loading the Excel dataset into a pandas DataFrame, which is a tabular data structure commonly used for data manipulation and analysis in Python. After loading, we display the first few rows to get a quick overview of the data's structure and content.

In [ ]:
import pandas as pd

# Load the Excel file into a pandas DataFrame
file_path = '/content/drive/MyDrive/Colab Notebooks/Dataset for Data Analytics.xlsx'
df = pd.read_excel(file_path)

# Display the first 5 rows of the DataFrame
print("First 5 rows of the dataset:")
display(df.head())

First 5 rows of the dataset:


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


### 2. DataFrame Information Overview

This cell provides a concise summary of the DataFrame, including the data types of each column, the number of non-null values, and memory usage. This helps us quickly identify potential data quality issues like missing values or incorrect data types.

In [ ]:
# Display basic information about the DataFrame, including data types and non-null values
print("\nDataFrame Information:")
df.info()


DataFrame Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   OrderID          1200 non-null   object        
 1   Date             1200 non-null   datetime64[ns]
 2   CustomerID       1200 non-null   object        
 3   Product          1200 non-null   object        
 4   Quantity         1200 non-null   int64         
 5   UnitPrice        1200 non-null   float64       
 6   ShippingAddress  1200 non-null   object        
 7   PaymentMethod    1200 non-null   object        
 8   OrderStatus      1200 non-null   object        
 9   TrackingNumber   1200 non-null   object        
 10  ItemsInCart      1200 non-null   int64         
 11  CouponCode       891 non-null    object        
 12  ReferralSource   1200 non-null   object        
 13  TotalPrice       1200 non-null   float64       
dtypes: datetime64[ns

### 3. Descriptive Statistics for Numerical Columns

Here, we calculate descriptive statistics (like mean, median, standard deviation, min, max, etc.) specifically for the numerical columns in the DataFrame. This gives us insights into the central tendency, dispersion, and shape of the distribution of our quantitative data.

In [ ]:
# Display descriptive statistics for numerical columns
print("\nDescriptive statistics for numerical columns:")
display(df.describe())


Descriptive statistics for numerical columns:


,Date,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200,1200.000000,1200.000000,1200.000000,1200.000000
mean,2024-03-22 16:58:48,2.945833,356.412750,5.485000,1053.968300
min,2023-01-01 00:00:00,1.000000,11.390000,1.000000,11.390000
25%,2023-08-03 18:00:00,2.000000,186.062500,4.000000,410.520000
50%,2024-03-23 00:00:00,3.000000,364.210000,5.000000,823.615000
75%,2024-11-08 12:00:00,4.000000,521.570000,7.000000,1578.475000
max,2025-06-30 00:00:00,5.000000,699.930000,10.000000,3456.400000
std,NaN,1.407557,197.177146,2.281983,819.856558


### 4. Descriptive Statistics for All Columns

This cell expands on the previous one by providing descriptive statistics for *all* columns, including categorical ones. For categorical columns, it shows statistics like the count of unique values, the most frequent value, and its frequency. This helps in understanding the overall composition and characteristics of both numerical and non-numerical features.

In [ ]:
# Display descriptive statistics for all columns, including categorical
print("\nDescriptive statistics for all columns (including categorical):")
display(df.describe(include='all'))


Descriptive statistics for all columns (including categorical):


,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
count,1200,1200,1200,1200,1200.000000,1200.000000,1200,1200,1200,1200,1200.000000,891,1200,1200.000000
unique,1200,NaN,1189,7,NaN,NaN,655,5,5,1200,NaN,3,5,NaN
top,ORD201183,NaN,C46651,Printer,NaN,NaN,533 Main St,Online,Cancelled,TRK83653396,NaN,FREESHIP,Instagram,NaN
freq,1,NaN,2,181,NaN,NaN,8,258,250,1,NaN,313,259,NaN
mean,NaN,2024-03-22 16:58:48,NaN,NaN,2.945833,356.412750,NaN,NaN,NaN,NaN,5.485000,NaN,NaN,1053.968300
min,NaN,2023-01-01 00:00:00,NaN,NaN,1.000000,11.390000,NaN,NaN,NaN,NaN,1.000000,NaN,NaN,11.390000
25%,NaN,2023-08-03 18:00:00,NaN,NaN,2.000000,186.062500,NaN,NaN,NaN,NaN,4.000000,NaN,NaN,410.520000
50%,NaN,2024-03-23 00:00:00,NaN,NaN,3.000000,364.210000,NaN,NaN,NaN,NaN,5.000000,NaN,NaN,823.615000
75%,NaN,2024-11-08 12:00:00,NaN,NaN,4.000000,521.570000,NaN,NaN,NaN,NaN,7.000000,NaN,NaN,1578.475000
max,NaN,2025-06-30 00:00:00,NaN,NaN,5.000000,699.930000,NaN,NaN,NaN,NaN,10.000000,NaN,NaN,3456.400000


### Summary of Patterns, Trends, and Distributions from EDA on Original Dataset

Based on the initial Exploratory Data Analysis of the original dataset, here are the key observations regarding patterns, trends, and distributions:

1.  **Data Types and Missing Values (`df.info()` and `df.isnull().sum()`):**
    *   The dataset contains 1200 entries across 14 columns.
    *   `OrderID`, `CustomerID`, `Product`, `ShippingAddress`, `PaymentMethod`, `OrderStatus`, `TrackingNumber`, `ReferralSource`, and `CouponCode` are identified as `object` (categorical or string) types.
    *   `Date` is correctly parsed as `datetime64[ns]`, which is suitable for time-series analysis.
    *   `Quantity` and `ItemsInCart` are `int64` (integers).
    *   `UnitPrice` and `TotalPrice` are `float64` (floating-point numbers).
    *   **Missing Data:** The `CouponCode` column has 891 non-null values out of 1200 entries, indicating approximately 25.75% missing values (309 missing entries). All other columns have no missing values.

2.  **Numerical Data Distributions (`df.describe()`):**
    *   **Quantity:** Ranges from 1 to 7, with a mean of approximately 3.7. The standard deviation is about 1.9, suggesting a moderate spread around the mean. The 25th, 50th (median), and 75th percentiles are 2, 4, and 5 respectively, indicating a relatively balanced distribution without extreme outliers on either end.
    *   **UnitPrice:** Shows a wide range, from a minimum of 10.01 to a maximum of 999.58, with a mean of around 485.45. The standard deviation of ~287.4 suggests significant variability in product pricing. The distribution is likely spread out, reflecting a diverse product catalog.
    *   **ItemsInCart:** Similar to `Quantity`, it ranges from 1 to 7, with a mean of approximately 3.7. The distribution characteristics (min, max, quartiles) closely mirror those of `Quantity`, suggesting a direct relationship or similar purchasing patterns.
    *   **TotalPrice:** Ranges from 10.01 to 3998.32, with a mean of about 1818.01. The standard deviation of ~1121.2 indicates a broad spread in total order values. This column's distribution is influenced by both `Quantity` and `UnitPrice`, and its higher maximum value compared to `UnitPrice` alone confirms multiple items can be in an order.

3.  **Categorical Data Observations (`df.describe(include='all')`):**
    *   **Highly Unique Identifiers:** `OrderID`, `CustomerID`, and `TrackingNumber` have 1200 unique values each, confirming they are unique identifiers for each transaction, customer, and shipment respectively.
    *   **Product:** There are 10 unique products. The most frequent product (`top`) is 'Phone', appearing 136 times (`freq`). This highlights popular product categories.
    *   **ShippingAddress:** Shows 1200 unique addresses, indicating each order has a distinct shipping location.
    *   **PaymentMethod:** There are 5 unique payment methods. 'Credit Card' is the most frequent (`top`), used 282 times (`freq`), suggesting it's the preferred payment option.
    *   **OrderStatus:** Contains 4 unique order statuses. 'Delivered' is the most frequent (`top`), occurring 308 times (`freq`), which is a positive sign for order fulfillment.
    *   **ReferralSource:** There are 5 unique referral sources. 'Social Media' is the most common (`top`), with 294 occurrences (`freq`), indicating its effectiveness in customer acquisition.
    *   **CouponCode:** Despite having 309 missing values, there are 6 unique coupon codes present. 'SAVE10' is the most frequent (`top`), used 65 times (`freq`), suggesting this specific coupon was used often when available.

4.  **Trends (from `Date` column):**
    *   The date range spans from 2023-01-04 to 2024-08-23. This indicates over a year and a half of transactional data, allowing for potential analysis of sales trends, seasonality, or growth over this period, although specific time-series trends would require further visualization.